src/Sdk/WebApi/WebApi/BrokerHttpClient.cs

```diff
@@ -172,5 +179,11 @@ public async Task DeleteSessionAsync(
 
             throw new Exception($"Failed to delete broker session: {result.Error}");
         }
+
+        protected static bool TryParseErrorBody(string errorBody, out BrokerError error)
+        {
+            var validator = new Func<BrokerError, bool>(e => e.Source != null && e.Source == "actions-broker-listener");
+            return RawHttpClientBase.TryParseErrorBody(errorBody, validator, out error);
```

Comment:
Consider copy/pasting [this impl](https://github.com/actions/runner/blob/36c66c808363839ea7069bef87fd4572c8bb6c54/src/Sdk/RSWebApi/RunServiceHttpClient.cs#L251) and avoiding callback indirection (arguably decreases readability) 🤷

Reply:
(the http client classes have a lot of layers already)